frontend/src/pages/Dashboard/index.tsx

```diff
@@ -55,104 +59,125 @@ const Dashboard: React.FC = () => {
     }
   };
 
-    useEffect(() => {
-        const fetchAssignments = async (semester: ISemester) => {
-            try {
-                if (semester.classroom_id) {
-                    const base_url: string = import.meta.env.VITE_PUBLIC_API_DOMAIN as string;
-                    const result = await fetch(`${base_url}/assignments/${semester.classroom_id}`, {
-                        method: 'GET',
-                        credentials: 'include',
-                        headers: {
-                            'Content-Type': 'application/json',
-                        },
-                    });
-
-                    if (!result.ok) {
-                        throw new Error('Network response was not ok');
-                    }
-
-                    const data: IAssignment[] = (await result.json() as IAssignment[])
-                    const assignmentGoodDate = data.map((assignment: IAssignment) => ({
-                        ...assignment,
-                        main_due_date: assignment.main_due_date ? new Date(assignment.main_due_date) : null,
-                    }))
-                    console.log("Setting Assignment data: ", assignmentGoodDate)
-                    setAssignments(assignmentGoodDate);
-                }
-
-            } catch (error: unknown) {
-                console.error('Error fetching assignments:', error);
-            }
-        };
-
-        const SyncWithClassroom = async (semester: ISemester) => {
-            try {
-                const base_url: string = import.meta.env.VITE_PUBLIC_API_DOMAIN as string;
-                const result = await fetch(`${base_url}/github/sync`, {
-                    method: 'POST',
-                    credentials: 'include',
-                    headers: {
-                        'Content-Type': 'application/json',
-                    },
-                    body: JSON.stringify({ classroom_id: semester.classroom_id }),
-                })
-
-                if (!result.ok) {
-                    throw new Error('Network response was not ok');
-                }
-
-            } catch (error: unknown) {
-                console.error('Error making API call:', error);
+  useEffect(() => {
+    const fetchAssignments = async (semester: ISemester) => {
+      try {
+        if (semester.classroom_id) {
+          const base_url: string = import.meta.env
+            .VITE_PUBLIC_API_DOMAIN as string;
+          const result = await fetch(
+            `${base_url}/assignments/${semester.classroom_id}`,
```

Comment:
needs to be updated to the new route for getting all assignments with a certain classroom id